coreweave/networking/resource_vpc_test.go

```diff
@@ -17,8 +21,91 @@ import (
 	"github.com/hashicorp/terraform-plugin-testing/plancheck"
 	"github.com/hashicorp/terraform-plugin-testing/statecheck"
 	"github.com/hashicorp/terraform-plugin-testing/tfjsonpath"
+
+	networkingv1beta1 "buf.build/gen/go/coreweave/networking/protocolbuffers/go/coreweave/networking/v1beta1"
+)
+
+const (
+	AcceptanceTestPrefix = "test-acc-"
 )
 
+func init() {
+	resource.AddTestSweepers("coreweave_vpc", &resource.Sweeper{
+		Name:         "coreweave_networking_vpc",
+		Dependencies: []string{},
+		F: func(r string) error {
+			ctx, cancel := context.WithTimeout(context.Background(), 30*time.Minute)
+			defer cancel()
+
+			testutil.SetEnvIfUnset(provider.CoreweaveApiTokenEnvVar, "test")
+			client, err := provider.BuildClient(ctx, provider.CoreweaveProviderModel{})
+			if err != nil {
+				return fmt.Errorf("failed to build client: %w", err)
+			}
+
+			listResp, err := client.ListVPCs(ctx, &connect.Request[networkingv1beta1.ListVPCsRequest]{})
+			if err != nil {
+				return fmt.Errorf("failed to list VPCs: %w", err)
+			}
+			for _, vpc := range listResp.Msg.Items {
+				if !strings.HasPrefix(vpc.Name, AcceptanceTestPrefix) {
+					log.Printf("skipping VPC %s because it does not have prefix %s", vpc.Name, AcceptanceTestPrefix)
+					continue
+				}
+
+				if vpc.GetZone() != r {
+					log.Printf("skipping VPC %s in zone %s because it does not match sweep zone %s", vpc.Name, vpc.Zone, r)
+					continue
+				}
+
+				log.Printf("sweeping VPC %s", vpc.Name)
+				if testutil.SweepDryRun() {
+					log.Printf("skipping VPC %s because of dry-run mode", vpc.Name)
+					continue
+				}
+
+				deleteReq := connect.NewRequest(&networkingv1beta1.DeleteVPCRequest{
+					Id: vpc.Id,
+				})
+				deleteResp, err := client.DeleteVPC(ctx, deleteReq)
+				if err != nil {
+					return fmt.Errorf("failed to delete VPC %s: %w", vpc.Name, err)
+				}
+				deletedVpc := deleteResp.Msg.Vpc
+
+				timeout := time.After(1 * time.Minute)
+				ticker := time.NewTicker(10 * time.Second)
+				defer ticker.Stop()
+				for {
```

Comment:
This is probably worth abstracting out into a function in the testutil package, a lot of resources will be asynchronous during their CRUD lifecycle - I'm thinking something that is similar to testify's [assert.Eventually](https://github.com/stretchr/testify/blob/v1.10.0/assert/assertions.go#L1930)